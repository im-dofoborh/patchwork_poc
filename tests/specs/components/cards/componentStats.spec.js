import { mount, createLocalVue } from '@vue/test-utils';
import VueRouter from 'vue-router';
import ComponentStatsCard from '~/components/cards/ComponentStats';

describe('ComponentStatsCard component', () => {
  let $stats = {
    getComponentStats() {
      return {
        componentCount: 1,
        variantCount: 2,
        atomCount: 3,
        moleculeCount: 4,
        organismCount: 5,
        templateCount: 6,
        todoCount: 7,
        mostUsedTags: {
          foo: 12,
          bar: 24,
        },
        leastUsedTags: {
          foo: 48,
          bar: 96,
        }
      }
    }
  };

  let localVue = null;
  let wrapper = null;
  let router = null;

  beforeEach(() => {
    localVue = createLocalVue();
    localVue.component('component-stats-card', ComponentStatsCard);
    localVue.use(VueRouter);
    router = new VueRouter();
    wrapper = mount(ComponentStatsCard, {
      localVue,
      router,
      mocks: {
        $stats,
      }
    });
  })

  it('renders default variant', () => {
    expect(wrapper.html()).toMatchSnapshot();
  });
});
