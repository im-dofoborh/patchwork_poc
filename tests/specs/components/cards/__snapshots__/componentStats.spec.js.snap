// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ComponentStatsCard component renders default variant 1`] = `
<section class="sg-component-stats-card">
  <div class="row">
    <div class="col-md-4 mb-md">
      <div class="sg-component-stats-card__section">
        <p class="heading-1"> 1 </p>
        <p class="body-copy-large mb-md"> Components </p>
        <p class="heading-1 mt-md"> 2 </p>
        <p class="body-copy-large"> Variants </p>
      </div>
    </div>
    <div class="col-md-4 mb-md">
      <div class="sg-component-stats-card__section">
        <p class="body-copy-large mb-md"> 3 Atoms </p>
        <p class="body-copy-large mb-md"> 4 Molecules </p>
        <p class="body-copy-large mb-md"> 5 Organisms </p>
        <p class="body-copy-large mb-md"> 6 Template </p>
        <p class="body-copy-large"> 7 ToDo </p>
      </div>
    </div>
    <div class="col-md-4 mb-md">
      <div class="sg-component-stats-card__section">
        <p class="body-copy-large mb-xs"> Most used tags: </p>
        <div class="mb-md">
          <div class="sg-component-stats-card__tag"><a href="#/components/list?search=foo" class="standard-tag standard-tag--link mb-xs">
              foo
            </a> <sup>12</sup></div>
          <div class="sg-component-stats-card__tag"><a href="#/components/list?search=bar" class="standard-tag standard-tag--link mb-xs">
              bar
            </a> <sup>24</sup></div>
        </div>
        <p class="body-copy-large mb-xs"> Least used tags: </p>
        <div>
          <div class="sg-component-stats-card__tag"><a href="#/components/list?search=foo" class="standard-tag standard-tag--link mb-xs">
              foo
            </a> <sup>48</sup></div>
          <div class="sg-component-stats-card__tag"><a href="#/components/list?search=bar" class="standard-tag standard-tag--link mb-xs">
              bar
            </a> <sup>96</sup></div>
        </div>
      </div>
    </div>
  </div>
</section>
`;
